backend/src/main/java/com/redhat/cloud/notifications/auth/kessel/KesselAuthorization.java

```diff
@@ -0,0 +1,132 @@
+package com.redhat.cloud.notifications.auth.kessel;
+
+import api.check.v1.CheckRequest;
+import api.check.v1.CheckResponse;
+import api.relations.v1.ObjectReference;
+import api.relations.v1.SubjectReference;
+import client.CheckClient;
+import com.redhat.cloud.notifications.auth.principal.ConsoleIdentity;
+import com.redhat.cloud.notifications.auth.principal.ConsolePrincipal;
+import com.redhat.cloud.notifications.auth.principal.rhid.RhIdentity;
+import com.redhat.cloud.notifications.auth.principal.rhid.RhServiceAccountIdentity;
+import com.redhat.cloud.notifications.config.BackendConfig;
+import io.quarkus.logging.Log;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.ForbiddenException;
+import jakarta.ws.rs.core.SecurityContext;
+
+import java.security.Principal;
+
+@ApplicationScoped
+public class KesselAuthorization {
+    /**
+     * Represents the "service account"'s subject's name in Kessel.
+     */
+    public static final String KESSEL_IDENTITY_SUBJECT_SERVICE_ACCOUNT = "service-account";
+    /**
+     * Represents the "user"'s subject's name in Kessel.
+     */
+    public static final String KESSEL_IDENTITY_SUBJECT_USER = "user";
+
+    @Inject
+    BackendConfig backendConfig;
+
+    @Inject
+    CheckClient checkClient;
```

Comment:
Shouldn't we rename this propoerty with something more explicit, like `kesselClient`?